src/agent/windows/CMDCommunication.cpp

```diff
@@ -0,0 +1,59 @@
+#include "CMDCommunication.h"
+#include <fstream>
+#include <iostream>
+
+GeneralHealth CMDCommunication::CollectDiskStatus()
+{
+    const std::string commandResult = ExecuteDiscStatusCommand();
+
+    m_generalHealth.SetStatus(GeneralHealth::Health::UNKNOWN);
+
+    if (commandResult == "OK")
+    {
+        m_generalHealth.SetStatus(GeneralHealth::Health::GOOD);
+    }
+    else if (commandResult == "Degraded")
+    {
+        m_generalHealth.SetStatus(GeneralHealth::Health::CHECK_STATUS);
+    }
+    else if (commandResult == "Pred Fail")
+    {
+        m_generalHealth.SetStatus(GeneralHealth::Health::BAD);
+    }
+
+    return m_generalHealth;
+}
+
+std::string CMDCommunication::ExecuteDiscStatusCommand() const
+{
+    std::string command = "wmic diskdrive get status";
+    std::string fileName = "dh.txt";

```

Comment:
There is a nice Qt class for creating custom dirs and files: QTemporaryDir and QTemporaryFile. Consider them.

Reply:
Also, w QProcess you can read process' output directly without a file